aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -233,12 +233,12 @@ static Tensor apply_bag_size(const Tensor &offsets, const Tensor &indices,
                              const int64_t mode, Tensor &output,
                              const Tensor &bag_size) {
   if (mode == MODE_MEAN) {
-    if (offsets.size(0) == 1) {
+    // Avoid dividing by 0 for empty bags.
+    // Instead we want empty bags to return all 0s
+    if (offsets.size(0) == 1 && indices.size(0) != 0) {
```

Comment:
This special casing is a bit hard to read. Could we just do `output /= std::max(bag_size_, 1)` ?

Reply:
Sorry I don't feel the same way :-)

Reply:
Another argument against the current condition is that if `offsets.size(0) == 1 && indices.size(0) == 0`, then we go to the other path. The other path creates a new tensor (via `at::ones` to do the max, which is really inefficient (uses ~3-6us) instead of a `std::max` call that will be much faster

Reply:
That being said, I don't think the performance of this is critical